linera-base/src/data_types.rs

```diff
@@ -856,7 +882,29 @@ impl<'a> Deserialize<'a> for HashedBlob {
     where
         D: Deserializer<'a>,
     {
-        Ok(Blob::deserialize(deserializer)?.into_hashed())
+        fn get_blob_id(blob_type: &BlobType, blob: &Blob) -> BlobId {
+            match blob_type {
+                BlobType::Data => BlobId::new_data(blob),
+            }
+        }
+
+        if deserializer.is_human_readable() {
+            let s = String::deserialize(deserializer)?;
+            let hashed_blob_bytes = hex::decode(s).map_err(serde::de::Error::custom)?;
+            let hashed_blob: SerializableBlob =
+                bcs::from_bytes(&hashed_blob_bytes).map_err(serde::de::Error::custom)?;
+
+            Ok(HashedBlob {
+                id: get_blob_id(&hashed_blob.blob_type, &hashed_blob.blob),
+                blob: hashed_blob.blob,
+            })
+        } else {
+            let hashed_blob = SerializableBlob::deserialize(deserializer)?;
+            Ok(HashedBlob {
+                id: get_blob_id(&hashed_blob.blob_type, &hashed_blob.blob),
+                blob: hashed_blob.blob,
+            })
+        }
```

Comment:
I guess `HashedBlob` is a bit of a misnomer now: It's a blob together with its ID (which contains its hash _and_ its blob type), and it's serialized as the blob together with its blob type.

Reply:
Yeah, I guess it's not just `hashed` now. Any name suggestions? 😅
I can't think of anything better than something like `BlobWithID` or something not great like that 😅

Reply:
Instead of `Blob` and `HashedBlob`, we could have `Blob` and `BlobWithId`, or even `BlobPayload` and `Blob`? (And I guess `SerializableBlob` could be `BlobWithType`, but I'm fine with `SerializableBlob`, too.)

Reply:
Ok! I'll go with `BlobPayload` and `Blob`

Reply:
This will be too big of a change though so I'll do it on a separate PR